docs/configuration/config-file-reference.md

```diff
@@ -5364,6 +5373,36 @@ time_window:
   # lookback delta) that the query should be within. If set to 0, it won't be
   # checked.
   [end: <int> | default = 0]
+
+# Limit that query time range should be within. If not set, it won't be checked.
+time_range_limit:
+  # Query time range should be above or equal to this value to match. If set to
+  # 0, it won't be checked.
+  [min: <int> | default = 0]
+
+  # Query time range should be below or equal to this value to match. If set to
+  # 0, it won't be checked.
+  [max: <int> | default = 0]
+
+# Limit that query step should be within. It will check subquery steps as well.
+# If not set, it won't be checked.
+query_step_limit:
```

Comment:
I am a little bit confused. Is this configuration to reject queries by step or use step to filter out queries to match the regex/dashboard ID or panel ID?

Reply:
This configuration is used to match queries. All the provided fields in QueryAttribute will use AND operand to decide if query matches or not. It does not directly reject or prioritize queries itself. The overall result of matching the QueryAttribute determines if a query will be rejected or prioritized, based on whether QueryAttribute is used within the QueryRejection or QueryPriority configuration respectively.

Example config:
```
    query_rejection:
      enabled: true
      query_attributes:
        - regex: .*ALERT.*
          query_step_limit:
            min: 6s
            max: 20s
          dashboard_uid: "dash123"
```
`curl 'localhost:8005/prometheus/api/v1/query?query=someCustomALERTquery&time=1718383304&end=1718386904&step=7s' -H "User-Agent: other" -H "X-Dashboard-Uid: dash123"`

In this case, all the queries who contains "ALERT" string AND step is '>= 6s AND <= 20s' AND dashboardUID is equal to "dash123" will be considered as matched query and as it's used in query_rejection, those matching queries will be rejected. If query fails to match any of them, then it will not be rejected. For example, if query is very similar to the one matched but has step of 30s, then it will not be rejected.
Matching is only done on a attributes that is provided in config. For example if query_step_limit config was not set in above query_attribute, then any query that match regex and dashboardUID would be considered as matched independent of the step size of the query and will be rejected.

Reply:
Updated PR description with this info as well.